app/models/conversation.rb

```diff
@@ -39,4 +40,26 @@ def self.share_conversation(user, match)
   def not_current_user(current_user)
     users.where.not(id: current_user.id).to_a
   end
+
+  def self.with_unread_messages_count_for(user_id)
+    Conversation
```

Comment:
I would move this to scope.
Also, when using a method like this one, it's better if you use `sefl.class` so if the name changes, you don't need to change here.